src/stack.c

```diff
@@ -0,0 +1,47 @@
+#include "stack.h"
+
+Stack *newStack(int size)
+{
+    Stack *stack = malloc(sizeof(Stack));
+    stack->top = -1;
+    stack->maxsize = size;
+    stack->elements = malloc(size * sizeof(uint16_t));
+    return stack;
+}
+void freeStack(Stack *st)
+{
+    free(st->elements);
+    free(st);
+}
+
+void push(Stack *st, uint16_t dir)
+{
+    if (!isFull(st))
+    {
+        st->elements[++st->top] = dir;
+    }
+}
+int pop(Stack *st)
+{
+    if (!isEmpty(st))
+    {
+        return st->elements[st->top--];
+    }
+    return -1;
```

Comment:
Maybe not the best approach as -1 is a valid value in memory (0xFF), we need some kind of exit signal there when an underflow happens. 

Leave it like that, I'll fill an issue and we will think about it later.